notebooks/standardize_events_data.ipynb

```diff
@@ -0,0 +1,201 @@
+    "1.  Load event data from Paraquet file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f8f05b8e-213b-47ef-9189-486d2e702267",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "# Define the GCS path to the Parquet file\n",
+    "gcs_path = 'https://storage.googleapis.com/derisk-persistent-state/zklend_data/events_sample.parquet'\n",
+    "\n",
+    "# Read the Parquet file into a Pandas DataFrame\n",
+    "df = pd.read_parquet(gcs_path, engine='pyarrow')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e9d58f1f-916d-4d2c-ba98-810ce9144846",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "62022959-dea1-48d1-a003-94f79c34bf43",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# taking a look at the entire second row\n",
+    "pd.set_option('display.max_colwidth', None)\n",
+    "df.iloc[1,:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1cdadf73-3237-47a4-acbd-e59abd6c9461",
+   "metadata": {},
+   "source": [
+    "2. Define a Function to Decode Byte Strings\n",
+    "Define a helper function to decode byte strings if necessary. This function will be used to ensure that data in the keys and data columns is correctly decoded before further processing."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "03358fb0-dd7b-4c13-a57a-4703a2622ecd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def decode_byte_string(value):\n",
+    "    if isinstance(value, bytes):\n",
+    "        return value.decode(\"utf-8\")\n",
+    "    return value"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9bde802d-8d13-48d3-a099-168f29c87400",
+   "metadata": {},
+   "source": [
+    "3. Define a Function to Transform Each Row\n",
+    "Define a function that decodes the keys and data columns, converts the hexadecimal strings to integers, and structures the information into a standardized format."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "86b8bd24-5ce1-431e-a0e2-f49d4056a34c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ast\n",
+    "\n",
+    "def decode_and_convert_row(row):\n",
+    "    # Decode 'keys' and 'data' fields\n",
+    "    keys_decoded = decode_byte_string(row['keys'])\n",
+    "    data_decoded = decode_byte_string(row['data'])\n",
+    "    \n",
+    "    # Convert string representation of list to actual list\n",
+    "    keys = ast.literal_eval(keys_decoded)\n",
+    "    data = ast.literal_eval(data_decoded)\n",
+    "    \n",
+    "    # Function to convert hexadecimal string to integer\n",
+    "    def hex_to_int(hex_str):\n",
+    "        try:\n",
+    "            return int(hex_str, 16)\n",
+    "        except ValueError:\n",
+    "            return hex_str\n",
```

Comment:
This function can't return the original string. Every value consumed needs to be converted to int.